src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowPathsSpec.groovy

```diff
@@ -38,4 +35,30 @@ class HaFlowPathsSpec extends HealthCheckSpecification {
         def actualException = thrown(HttpClientErrorException)
         new HaFlowNotFoundExpectedError(~/HA-flow ${flowId} not found\./).matches(actualException)
     }
+
+    @Tidy
+    def "HA flow main path is not overlapped with protected path"() {
+        assumeTrue(useMultitable, "HA-flow operations require multiTable switch mode")
+        given: "An HA-flow with protected path"
+        def swT = topologyHelper.findSwitchTripletForHaFlowWithProtectedPaths()
+        assumeTrue(swT != null, "These cases cannot be covered on given topology:")
+
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT)
+        haFlowRequest.allocateProtectedPath = true
+        HaFlowExtended haFlow = swT.addHaFlow(haFlowRequest)
+        assert haFlow.allocateProtectedPath
+
+        when: "Request path for HA flow with protected path"
+        def haFlowPaths = haFlow.getAllEntityPaths()
+        assert haFlowPaths.subFlowPaths.protectedPath.size() == 2
```

Comment:
This check is not necessary if you verify that each subflow path has protected path.

Reply:
I added this check` haFlowPaths.subFlowPaths.protectedPath.size() == 2 `
to clarify that Ha flow has a protected path.  The following check (`getCommonIsls()`) will be skipped as we don't have entities to compare (decreasing time execution).